File: competitive-programming/atcoder/abc227/c.c
#include <stdio.h>
#include <math.h>

int main()
{
  long long n;
  scanf("%lld", &n);
  long long result = 0;
  for (long long a = 1; a <= (long long)floor(pow(n, 1.0/3.0)); ++a) {
    for (long long b = a; b <= (long long)floor(sqrt((double)n/a)); ++b) {
      result += (long long)floor((double)n/(a*b)) - b + 1;
    }
  }
  printf("%lld\n",result);
  return 0;
}

File: competitive-programming/atcoder/abc227/c.py
from math import floor, pow, sqrt

def main():
    n = int(input())
    # n = 4
    result = 0
    for a in range(1, floor(pow(n, 1/3)) + 1):
        for b in range(a, floor(sqrt(n/a)) + 1):
            result += floor(n/(a*b)) - b + 1
    print(result)

main()
